src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,47 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception ex) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t insert phone " + phone, ex);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            List<Predicate> andPredicates = new ArrayList<>();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                String key = entry.getKey();
+                String[] values = entry.getValue();
+                List<Predicate> orPredicates = new ArrayList<>();
+                for (String value : values) {
+                    orPredicates.add(criteriaBuilder.equal(root.get(key), value));
+                }
+                andPredicates.add(criteriaBuilder.or(orPredicates.toArray(new Predicate[]{})));
+            }
+            Predicate allPredicates = criteriaBuilder.and(andPredicates.toArray(new Predicate[]{}));
```

Comment:
This algorithm can be simplified by using `.in()` method, which takes `varargs` values (this solution is valid, but possible improvements are up to you).